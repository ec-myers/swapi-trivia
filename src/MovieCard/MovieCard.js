import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types'
import './MovieCard.scss';

const MovieCard = ({movie, goToMovieCharacters}) => {
  return(
    <div className="Movie-Card">
      <h2>{movie.title}</h2>
      <h3>Episode: {movie.episode}</h3>
      <h3>Released in {movie.releaseYear}</h3>
      <Link onClick={goToMovieCharacters}  to={`/movies/${movie.id}`}>
          <button id={movie.id}>Explore Characters</button> 
      </Link>
    </div>
  )
}

export default MovieCard; 

MovieCard.propTypes = {
  movie: PropTypes.object.isRequired,
  goToMovieCharacters: PropTypes.func.isRequired

}

