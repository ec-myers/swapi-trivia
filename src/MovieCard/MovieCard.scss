$accent-color-dark: rgba(108, 91, 117, 0.8);
$accent-color-light: rgba(242,136,79, 0.8);
$hover-color: rgba(231,101,81, 0.8);
$font-color: #ffffff;

.Movie-Card {
  align-items: center;
  background-color: $accent-color-dark;
  border-radius: 10px;
  border: solid 2px #ffffff;
  box-shadow: 0 0 5px #ffffff;
  box-sizing: border-box;
  color: $font-color;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 20px;

  h2 {
    border-bottom: 2px solid #ffffff;
    font-size: 28px;
  }

  h3 {
    font-size: 24px;
  }

  button {
    border-radius: 5px;
    background: $accent-color-light;
    color: $font-color;
    height: 40px;
    width: 225px;
    font-size: 18px;

    &:hover {
      cursor: pointer;
      background-color: $hover-color;
    }
  }
}