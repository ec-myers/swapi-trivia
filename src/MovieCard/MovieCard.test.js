import React from 'react'
import { shallow } from 'enzyme'
import MovieCard from './MovieCard'

describe('MovieCard', () => {
  let wrapper, mockMovie, mockGoToMovieCharacters;
  beforeEach(() => {
    mockGoToMovieCharacters = jest.fn()
    mockMovie = {
      title: 'Bad Prequel',
      episode: 1,
      releaseYear: 2001,
      id: 3
    }

    wrapper = shallow(<MovieCard movie={mockMovie} goToMovieCharacters={mockGoToMovieCharacters} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call goToMovieCharacters on click', () => {
    wrapper.find('Link').simulate('click')
    
    expect(mockGoToMovieCharacters).toHaveBeenCalled()
  })
})

