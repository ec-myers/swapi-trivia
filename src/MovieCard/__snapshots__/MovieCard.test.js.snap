// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
<div
  className="Movie-Card"
>
  <h2>
    Bad Prequel
  </h2>
  <h3>
    Episode: 
    1
  </h3>
  <h3>
    Released in 
    2001
  </h3>
  <Link
    onClick={[MockFunction]}
    to="/movies/3"
  >
    <button
      id={3}
    >
      Explore Characters
    </button>
  </Link>
</div>
`;
